src/app/client/src/app/modules/dashboard/components/data-chart/data-chart.component.ts

```diff
@@ -398,13 +399,22 @@ export class DataChartComponent implements OnInit, OnDestroy {
   @Input()
   set globalFilter(val: any) {
     if(val){
-      val.chartData['selectedFilters'] = {};
-      this.chartData = val.chartData;
-      this.chartData['selectedFilters'] = { };
-      this.cdr.detectChanges();
-      this.getDataSetValue(val.chartData);
-      this.calculateBigNumber(val.chartData);
-      this.resetForm();
+      let updatedChartData = val.chartData.filter(chart=>{
+        if(this.chartConfig['id']==chart.id){
+          return chart.data;
+        }
+      });
+
+      if(updatedChartData){
```

Comment:
check should be on whether the first element of the array is present or not .. because even the empty array returns truthy

Reply:
Also all proper interfaces/types